extensions/amp-story/0.1/bookend-share.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {ICONS} from './icons';
+import {Services} from '../../../src/services';
+import {isObject} from '../../../src/types';
+import {createElementWithAttributes, escapeHtml} from '../../../src/dom';
+import {dev, user} from '../../../src/log';
+import {dict} from './../../../src/utils/object';
+
+
+const LINK_SHARE_TEMPLATE =
+    `<div class="i-amphtml-story-share-icon">
+      ${ICONS.link}
+    </div>
+    <span class="i-amphtml-story-share-name">Get link</span>`;
+
+
+/**
+ * Maps share provider type to visible name.
+ * If the name only needs to be capitalized (e.g. `facebook` to `Facebook`) it
+ * does not need to be included here.
+ * @const {!JsonObject}
+ */
+const SHARE_PROVIDER_NAME = dict({
+  'gplus': 'Google+',
+  'linkedin': 'LinkedIn',
+  'whatsapp': 'WhatsApp',
+});
+
+
+/**
+ * @param {string} type
+ * @param {!JsonObject=} opt_params
+ * @return {string}
+ */
+// TODO(alanorozco): article metadata
+function shareProviderHtml(type, opt_params) {
+  const params = !opt_params ? '' :
+      Object.keys(opt_params)
+          .map(field =>
+              `data-param-${escapeHtml(field)}=` +
```

Comment:
DOM API. Then you won't need escapeHtml either.

Reply:
Done